src/main/java/DukeException.java

```diff
@@ -0,0 +1,80 @@
+public class DukeException extends Exception {
+
+    protected DukeExceptionType exceptionType;
+    protected Commands command;
+
+    /**
+     * constructor for a DukeException object
+     * @param e error name
+     * @param exceptionType type of error
+     */
+    public DukeException(String e, DukeExceptionType exceptionType){
+        super(e);
+        this.exceptionType = exceptionType;
+    }
+
+    /**
+     * constructor for a DukeException object
+     * @param e error name
+     * @param exceptionType type of error
+     * @param command the command which results in the error
+     */
+    public DukeException(String e, DukeExceptionType exceptionType, Commands command){
+        super(e);
+        this.exceptionType = exceptionType;
+        this.command = command;
+    }
+
+    @Override
+    public String toString(){
+        String error = "";
+        switch (exceptionType){
+            case EMPTY_TIME:
+                switch (command){
+                    case DEADLINE:
+                        error+= "Deadline task requires a time";
+                        break;
+                    case EVENT:
+                        error+= "Event task requires a time";
+                        break;
+                }
+                break;
+            case INVALID_TASK:
+                error+= "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+                break;
+            case NO_DESCRIPTION:
+                switch (command){
+                    case DEADLINE:
+                        error+= "Deadline task description cannot be empty";
```

Comment:
Would it be better if the indentation of the case statements follow that of the coding standard?